cmd/containerd-shim-runhcs-v1/task_hcs.go

```diff
@@ -255,6 +259,13 @@ func newHcsTask(
 		}); err != nil {
 		return nil, err
 	}
+
+	// wait for event notifications and publish them; goroutine will exit with
+	// ctx, or when container is closed.
+	// event publication does not have an event time stamp, so start this after
+	// after publishing creation event
```

Comment:
why does this matter?

Reply:
if the container (almost) immediately OOMs after starting, and the notification go routine (`publishNotifications`) is started before the `ht.events.publishEvent` call publishes the start event for the container, then containerd and others could see that an OOM event happened before the container started (depending on scheduling), which would be weird sequencing of events.

Reply:
Does this mean we may miss an OOM event though?

Reply:
nope, the OOM events are queued, `publishNotifications` just dequeues and forwards them to containerd